manifests/params.pp

```diff
@@ -171,6 +171,30 @@
       $psql_path              = pick($psql_path, '/usr/bin/psql')
     }
 
+    'Gentoo': {
+      $user                = pick($user, 'postgres')
+      $group               = pick($group, 'postgres')
+
+      $client_package_name  = pick($client_package_name, 'UNSET')
```

Comment:
To answer the easy question first: the `$server_package_name` within the pick call is resolving to `$::postgresql::globals::server_package_name` as that is inherited into this class. The assignment happens in the local scope, setting `$::postgresql::params::server_package_name`, hiding the value from the globals. This chaining allows the user to override values at any of the levels they desire.

The issue with using `undef` is that it will reset an argument to the parameter default. That's why you need to use something like `UNSET`.